src/components/GoalCriteria/GoalCriteria.tsx

```diff
@@ -290,7 +290,8 @@ interface GoalCriteriaProps {
     goalId: string;
     criteriaList?: GoalAchiveCriteria[];
     canEdit: boolean;
-    onClick?: (itenm: GoalAchiveCriteria) => void;
+    canCreate: boolean;
```

Comment:
Component must be refactored. It's better to use composition instead. U can pass or not smth to render via props instead of using props as condition